code/game/gamemodes/objective.dm

```diff
@@ -38,6 +41,8 @@
 			possible_targets += possible_target
 	if(possible_targets.len > 0)
 		target = pick(possible_targets)
+	else
+		target = null//we'd rather have no target than an invalid one
```

Comment:
This proc was never used to assign a new target outside of what happens when the objective is created. This is safe.